uniffi_macros/src/export/attributes.rs

```diff
@@ -249,57 +250,85 @@ impl ExportedImplFnAttributes {
     pub fn new(attrs: &[Attribute]) -> syn::Result<Self> {
         let mut this = Self::default();
         for attr in attrs {
-            let segs = &attr.path().segments;
-
-            let fst = segs
-                .first()
-                .expect("attributes have at least one path segment");
-            if fst.ident != "uniffi" {
-                continue;
-            }
-            ensure_no_path_args(fst)?;
-
-            let args = match &attr.meta {
-                Meta::List(_) => attr.parse_args::<ExportFnArgs>()?,
-                _ => Default::default(),
-            };
-            this.args = args;
-
-            if segs.len() != 2 {
-                return Err(syn::Error::new_spanned(
-                    segs,
-                    "unsupported uniffi attribute",
-                ));
-            }
-            let snd = &segs[1];
-            ensure_no_path_args(snd)?;
-
-            match snd.ident.to_string().as_str() {
-                "constructor" => {
-                    if this.constructor {
-                        return Err(syn::Error::new_spanned(
-                            attr,
-                            "duplicate constructor attribute",
-                        ));
+            let path = attr.path();
+
+            if is_uniffi_path(path) {
+                this.process_path(path, attr, &attr.meta)?;
+            } else if matches!(attr.meta, Meta::List(_)) {
```

Comment:
Maybe this should also check that `attr.meta.path` is `cfg_attr`?  It seems safer to me to specifically target that attribute rather than all list-like attributes.

Reply:
I think because `uniffi::constructor` and `uniffi::method` are just markers, it's safe to ignore external APIs and assume that the user's intention, in the context of **uniffi** itself, is to mark the current method as one of the two variants, regardless of external conditions.

We don't need to match special cases like `cfg_attr`. It doesn't really matter if we match `uniffi::constructor`, `cfg_attr(whatever, uniffi::constructor)` or `foo(bar(baz(uniffi::constructor)))` from some `foobar` crate because the external wrappers are not part of our API, and users are free to mess with them as they need.

This general-purpose solution keeps our current API intact, so there's no need to document special cases or create new syntax for conditional parsing.

Reply:
I tend to agree with @rafaelbeckel, but if anybody feels really strong about limiting the implementation to only work with `cfg_attr`, I'm happy to make a change. After all, my main goal is to get it working at least for `cfg_attr`.

Reply:
I don't feel strongly about this one.  It feels safer to me, but I'm struggling to come up with an example where it would matter one way or another.